core/new-gui/src/app/workspace/component/visualization-panel-content/visualization-panel-content.component.ts

```diff
@@ -85,7 +97,12 @@ export class VisualizationPanelContentComponent implements AfterViewInit, OnDest
 
     this.data = result.table as object[];
     this.chartType = result.chartType;
-
+    if (!this.data || !this.chartType) {
+      return;
+    }
+    if (this.data?.length < 1) {
+      return;
+    }
```

Comment:
in extreme cases, if the data returned is supposed to be empty or null, shouldn't we still display an empty map or an empty chart? it seems like the current logic skips the rendering of the map/chart in such extreme cases.

Reply:
that will be handled in the next PR because I will render the map anyways in the background in the display result. This PR is depending on initializing the popup window, so it should be fine

Reply:
In that case, It will be good to left comments/TODOs in the code, so we know what to expect. Ideally you should mention this is a partial PR and give a step number to in the PR description and title. You can refer to word cloud PRs.